src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -234,6 +240,23 @@ private void WatchForFileSystemEventsThreadStart(object arg)
                 {
                     bool started = Interop.EventStream.FSEventStreamStart(_eventStream);
 
+                    _rootHandle = Interop.Sys.Open(_fullDirectory, Interop.Sys.OpenFlags.O_RDONLY, 0);
+                    if (_rootHandle.IsInvalid)
+                    {
+                        throw new FileNotFoundException(SR.Format(SR.FSW_IOError, _fullDirectory));
```

Comment:
Also, we probably still want to Dispose of the handle, even though it's invalid, e.g.
```C#
SafeFileHandle rootHandle = Interop.Sys.Open(_fullDirectory, Interop.Sys.OpenFlags.O_RDONLY, 0);
if (rootHandle.IsInvalid)
{
    rootHandle.Dispose();
    throw new FileNotFoundException(SR.Format(SR.FSW_IOError, _fullDirectory);
}
_rootHandle = rootHandle;